src/hdnode.js

```diff
@@ -206,6 +206,15 @@ HDNode.prototype.toBase58 = function (__isPrivate) {
 
 // https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#child-key-derivation-ckd-functions
 HDNode.prototype.derive = function (index) {
+  typeforce('Number', index)
+  var highestIndex = (HDNode.HIGHEST_BIT * 2) // should be 0x100000000
+  if (index < 0) {
+    throw new Error('Could not derive with negative index')
+  }
+  if (index >= highestIndex) {
+    throw new Error('Could not derive with too high index')
+  }
+
```

Comment:
@runn1ng it might be easiest to make a custom [typeforce](https://github.com/dcousens/typeforce) type for this :)